### ISSUE TITLE
copy functionality broken on Windows

### ISSUE THREAD

**Issue by screne** [geany/geany#1971]
I have Geany 1.33 installed on Windows 10.

When I copy text in Geany via context menu and paste it in Geany, it works.
When I copy text in Geany via ctrl-c and paste it in Geany, it works.
When I copy text in Geany via context menu and paste it outside of Geany, it works.
When I copy text in Geany via ctrl-c and paste it outside of Geany, it DOESN'T always work.

I didn't experience this behavior with any other application installed on my Windows 10 system.
I also didn't experience this behavior with a Xubuntu system I'm using.

**Comment by elextr** [comment]
> When I copy text in Geany via ctrl-c and paste it outside of Geany, it DOESN'T always work.

It would be useful to identify when it works and when it doesn't.

**Comment by codebrainz** [comment]
@screne is it by any chance the same issue as #1945?

**Comment by screne** [comment]
> 
> 
> > When I copy text in Geany via ctrl-c and paste it outside of Geany, it DOESN'T always work.
> 
> It would be useful to identify when it works and when it doesn't.

I haven't found a pattern yet. When I start Geany it works. After some time, switching through and copying from different applications, it suddenly stops working.

**Comment by screne** [comment]
> 
> 
> @screne is it by any chance the same issue as #1945?

Could be the same issue.

**Comment by adabru** [comment]
> > @screne is it by any chance the same issue as #1945?
> 
> Could be the same issue.

@screne could you verify/falsify whether the issue described in #1945 is affecting you? For example by stopping all running VirtualBox instances and check whether the issue still persists? And by checking whether the issue also occurs in MS Excel?

**Comment by screne** [comment]
I don't use MS Excel. So, I cannot test that. But I use VirtualBox. Stopping all VMs and closing VirtualBox seems to have a positive effect. At least a short test showed that ctrl-c was working again. But I don't know how reliable that conclusion is. It needs more investigation.

**Comment by Enverex** [comment]
Just to add to this (as I'm running into the same problem) - it's incredibly erratic. Copy+Paste will work reliably within Geany itself, but pasting out to external problems doesn't always fail, just most of the time (it may work fine a few times in a row, and then just not work the next 10 times you try).

Windows 8 with VirtualBox running here, so it looks like VirtualBox is the common denominator (especially with https://www.virtualbox.org/ticket/16192). Unfortunately there's been nothing but radio-silence on VirtualBox's side for over 3 months now.

**Comment by rolfschumacher** [comment]
Just to add to this(as I'm running into the same problem)
I ran Win10 and do not have VirtualBox running on this company laptop.
 I can confirm @screne , that on restarting Geany ctrl-c copying works for a while.

**Comment by elextr** [comment]
To all who are experiencing this, as no Geany developers are using Windows often enough to find it you need to provide more information.  What other apps are running? Did you use one of those just before the error? What did you do?

**Comment by AragCF** [comment]
From time to time Geany fails to copy the text to clipboard on Windows.
Even Ctrl+X doesn't help.
The clipboard gets empty and then no text is passed to the clipboard.
Sometimes in other programs I also get the error with clipboard.
I suggest the developers to make an "for" loop to try for, let's say, 10 times, to copy the text (save it to clipboard). I think you should get the error status from Windows API.
A am a web (php + JS) and VB6 programmer and under VB6 I do as I described -- it works fine for me.

I hope this helps.

**Comment by elextr** [comment]
@AragCF just to be clear, the code that copies text to the clipboard is part of the Scintilla editor widget Geany uses, and Scintilla uses the GTK GUI library to do the copy, so its not something Geany has much influence over.

Somebody could try building Geany on windows with the GTK3 version and see if it changes anything.

**Comment by AragCF** [comment]
Thanks you @elextr for the explanation.
I can build Geany on Windows using GTK3 if you give me the step-by-step instructions.

**Comment by elextr** [comment]
@eht16 do such instructions for windows exist?

**Comment by codebrainz** [comment]
Aside from packaging/installer, it's basically identical to compiling on Linux.

**Comment by AragCF** [comment]
Hm OK let's give it a try. Thank you. I'll come back with the news.

**Comment by eht16** [comment]
@AragCF you can find build instructions here: https://wiki.geany.org/howtos/win32/msys2

**Comment by AragCF** [comment]
> @AragCF you can find build instructions here: https://wiki.geany.org/howtos/win32/msys2

Thanks so much!

**Comment by AragCF** [comment]
Hey guys. I discovered that restarting Geany solves the problem. But the bug seems to be existing anyways.

**Comment by practicafz** [comment]
I have a similar problem with Xiphos on Windows (it is a GTK app). I just found out that if I hit Ctrl-C FOUR TIMES in a row, the paste will work outside of Xiphos. Not sure if this will apply for Geany.

**Comment by AragCF** [comment]
> I have a similar problem with Xiphos on Windows (it is a GTK app). I just found out that if I hit Ctrl-C FOUR TIMES in a row, the paste will work outside of Xiphos. Not sure if this will apply for Geany.

Also applies. True. But the qty of times is varying.

**Comment by practicafz** [comment]
True. I found out after posting that it is variable. Sometimes you have to hammer it 8 or more times. My son thought it might be a problem with the CTRL message getting separated from the C key message, when they are being passed to Windows. He tried hitting them both at the same time, and it worked a few times, but was not easily reproduceable either.
